Adjust naming: compoundAttribute -> structFieldAttribute.

toregge opened the pull request:
@geirst: please review

Commit:
Adjust naming: compoundAttribute -> structFieldAttribute.

geirst commented on searchcore/src/vespa/searchcore/proton/attribute/attribute_writer.h:
```diff
@@ -27,20 +27,20 @@ class AttributeWriter : public IAttributeWriter
         const FieldPath &getFieldPath() const { return _fieldPath; }
         void buildFieldPath(const DocumentType &docType);
-        bool getCompoundAttribute() const { return _compoundAttribute; }
+        bool getStructFieldAttribute() const { return _structFieldAttribute; }
```
Consider rename to isStructFieldAttribute()

geirst commented on searchcore/src/vespa/searchcore/proton/attribute/attribute_writer.h:
```diff
@@ -50,12 +50,12 @@ class AttributeWriter : public IAttributeWriter
         uint32_t getExecutorId() const { return _executorId; }
         const std::vector<WriteField> &getFields() const { return _fields; }
-        bool getHasCompoundAttribute() const { return _hasCompoundAttribute; }
+        bool getHasStructFieldAttribute() const { return _hasStructFieldAttribute; }
```
Consider rename to hasStructFieldAttribute()

geirst commented on searchcore/src/vespa/searchcore/proton/attribute/attribute_writer.h:
```diff
@@ -93,7 +93,7 @@ class AttributeWriter : public IAttributeWriter
 
     void onReplayDone(uint32_t docIdLimit) override;
-    bool getHasCompoundAttribute() const override;
+    bool getHasStructFieldAttribute() const override;
```
Consider rename to hasStructFieldAttribute()

geirst commented on searchcore/src/vespa/searchcore/proton/attribute/i_attribute_writer.h:
```diff
@@ -66,7 +66,7 @@ class IAttributeWriter {
     virtual void onReplayDone(uint32_t docIdLimit) = 0;
 
-    virtual bool getHasCompoundAttribute() const = 0;
+    virtual bool getHasStructFieldAttribute() const = 0;
```
Consider rename to hasStructFieldAttribute()

geirst reviewed (approved):
Only nitpicks regarding function names

Commit:
Simplify method names based on review feedback:
  getHasStructFieldAttribute() => hasStructFieldAttribute()
  getStructFieldAttribute()    => isStructFieldAttribute()

geirst reviewed (approved):
👍
